node/src/components/consensus/protocols/highway.rs

```diff
@@ -356,6 +356,40 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
     pub(crate) fn median_round_exp(&self) -> Option<u8> {
         self.highway.state().median_round_exp()
     }
+
+    fn drop_dependent_vertices<It>(&mut self, vertices: It)
+    where
+        It: IntoIterator<Item = Dependency<C>>,
+    {
+        // collect the vertices that depend on the ones we got in the argument and their senders...
+        let (senders, dependent_vertices): (HashSet<I>, Vec<Dependency<C>>) = vertices
+            .into_iter()
+            .flat_map(|vertex| self.vertex_deps.remove(&vertex))
+            .flatten()
+            .map(|(sender, pvv)| (sender, pvv.inner().id()))
+            .unzip();
+        // ...and drop them, too
+        self.drop_dependent_vertices(dependent_vertices);
+        // also, drop all vertices from the senders - if they sent us something dependent on an
+        // invalid vertex, they can't be trusted
+        self.drop_by_senders(senders);
+    }
+
+    fn drop_by_senders(&mut self, senders: HashSet<I>) {
+        let mut dropped_vertices = Vec::new();
+        for dependent_vertices in self.vertex_deps.values_mut() {
+            dependent_vertices.retain(|(sender, pvv)| {
+                if senders.contains(sender) {
+                    // remember the deps we drop
+                    dropped_vertices.push(pvv.inner().id());
+                    true
```

Comment:
Isn't that the wrong way around?

Reply:
What do you mean? This drops the vertices from anyone in `senders` that still await some dependencies. We remember which ones we drop, so that we can still drop ones that await _these_ vertices as _their_ dependencies.

Reply:
Ah, nevermind - you mean it should be `false` - of course :man_facepalming: